app/assets/javascripts/discourse/app/lib/autocomplete.js

```diff
@@ -374,21 +373,26 @@ export default function (options) {
     } else {
       selectedOption = -1;
     }
-    ul.find("li").click(function ({ originalEvent }) {
+    ul.find("li").click(async function ({ originalEvent }) {
+      // this is required to prevent the default behaviour when clicking on a <a> tag
+      originalEvent.preventDefault();
```

Comment:
Had to `preventDefault()` otherwise the page would reload if you were to click on a autocomplete item 😬

Reply:
Yeah, previously the function was returning `false` -  which is jQuery for "prevent the default" but now we're returning a promise so this no longer works.

which also means you can remove the `return false` at the bottom (line 400) 🙂